report/app/report/models/ExtendedNotificationReport.scala

```diff
@@ -0,0 +1,105 @@
+package report.models
+
+import java.time.Instant.ofEpochMilli
+import java.time.LocalDateTime
+import java.time.LocalDateTime.ofInstant
+import java.time.ZoneOffset.UTC
+import java.util.UUID
+
+import azure.NotificationDetails
+import com.gu.notifications.events.model._
+import models._
+import org.joda.time.DateTime
+import play.api.libs.json.{Format, Json}
+import play.api.libs.json.JodaWrites._
+import play.api.libs.json.JodaReads._
+
+case class ExtendedEventAggregation(
+  platformCounts: PlatformCount,
+  providerCounts: ProviderCount,
+  timing: Option[List[(LocalDateTime, Int)]]
+)
+
+object ExtendedEventAggregation {
+  implicit val extendedEventAggregationJF: Format[ExtendedEventAggregation] = Json.format[ExtendedEventAggregation]
+
+  def full(dynamoEventAggregation: DynamoEventAggregation, originalSentTime: LocalDateTime): ExtendedEventAggregation = {
+    def toOrderedTiming(timing: List[List[Int]]): Option[List[(LocalDateTime, Int)]] = {
+      val sentTime = originalSentTime.truncatedTo(TenSecondUnit)
+      val (_, orderedTiming) = dynamoEventAggregation.timing.map(timed => (timed.head, timed(1))).foldLeft((sentTime, List[(LocalDateTime, Int)]())) {
+        case ((lastTime, newList), (offset, count)) =>
+          val nextTime = lastTime.plusSeconds(10 * offset)
+          (nextTime, newList :+ (nextTime, count))
```

Comment:
This should probably use prepend like:
```((nextTime, count) :: nextTime, newList)```
finally reverse if needed.
There's a big performance penalty when using `:+`